### PR TITLE
[cli] support reading individual operational dataset values

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

**Comment by ndrake** [comment]
👍

File: src/cli/cli_dataset.cpp [hunk 1]
```diff
@@ -445,14 +550,29 @@ otError Dataset::ProcessMeshLocalPrefix(uint8_t aArgsLength, char *aArgs[])
...
+    if (aArgsLength == 0)
+    {
+        if (sDataset.mComponents.mIsNetworkNamePresent)
+        {
+            mInterpreter.mServer->OutputFormat("%.*s\r\n", static_cast<uint16_t>(sizeof(sDataset.mNetworkName)),
+                                               sDataset.mNetworkName.m8);
+        }
+        else
+        {
+            mInterpreter.mServer->OutputFormat("none\r\n");
```
Comment (simonlingoogle) [inline src/cli/cli_dataset.cpp:563]: How can we tell if there is no `Network Name` in dataset, or the `Network Name` is "none" ?

Comment (simonlingoogle) [reply src/cli/cli_dataset.cpp:563]: I would prefer (slightly) not to output anything if the corresponding field is not present in the dataset (not just `networkname`, but all fields)
Thoughts? @jwhui

File: src/cli/cli_dataset.cpp [hunk 2]
```diff
@@ -98,7 +98,7 @@ otError Dataset::Print(otOperationalDataset &aDataset)
 
     if (aDataset.mComponents.mIsChannelMaskPresent)
     {
-        mInterpreter.mServer->OutputFormat("Channel Mask: %08x\r\n", aDataset.mChannelMask);
+        mInterpreter.mServer->OutputFormat("Channel Mask: 0x%08x\r\n", aDataset.mChannelMask);
```
Comment (wgtdkp) [inline src/cli/cli_dataset.cpp:101]: Although `Channel Mask` is implemented as an `uint32_t`, but it is defined as a bit mask with variable length. Do we have a need to make this change?

Comment (jwhui) [reply src/cli/cli_dataset.cpp:101]: Thread 1.2 Section 8.10.1.18.1 defines the Channel Mask Entry for IEEE 802.15.4-2006 2.4 GHz PHY as a 4 byte value. If a new Channel Page is defined in the future, we will need to more more changes than just the Channel Mask. Thoughts?

Comment (wgtdkp) [reply src/cli/cli_dataset.cpp:101]: okay, I am fine with it.

**Pull request merged**